File: Practice(Beginner)/C++/Cutting Recipes.cpp
// Source: https://www.codechef.com/problems/RECIPE

#include <bits/stdc++.h>
using namespace std;

int main() {

    long long int t, n, GCD;
    cin >> t;
    while (t--){
        cin >> n;
        long long int arr[n];
        for (int i = 0; i < n; ++i) {
            cin >> arr[i];
        }
        GCD = __gcd(arr[0],arr[1]);
        for (int i = 2; i < n; ++i) {
            GCD = __gcd(GCD,arr[i]);
        }
        for (int i = 0; i < n; ++i) {
            cout << arr[i]/GCD << " ";
        }
        cout << endl;
    }

    return 0;
}

File: Practice(Beginner)/Python3/Cutting_Recipes.py
# Source: https://www.codechef.com/problems/RECIPE

import math
from functools import reduce
n = int(input())
for i in range(n):
    l = list(map(int,input().split()))
    x = reduce(math.gcd,l[1:])
    for j in l[1:]:
        print(j//x,end=" ")
    print()
